compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp

```diff
@@ -20,20 +20,21 @@ static __sanitizer::atomic_uintptr_t caller_pcs[kMaxCallerPcs];
 // that "too many errors" has already been reported.
 static __sanitizer::atomic_uint32_t caller_pcs_sz;
 
-__attribute__((noinline)) static bool report_this_error(uintptr_t caller) {
+SANITIZER_INTERFACE_WEAK_DEF(void, __ubsan_report_error, uintptr_t caller,
```

Comment:
maybe sinc into __ubsan_report_error
```
    char msg_buf[MSG_BUF_LEN(msg)] = MSG_TMPL(msg);              \
    decorate_msg(MSG_TMPL_END(msg_buf, msg), caller);            
```

Reply:
and we can remove decorated_msg

Reply:
We can't because it uses compile time string lengths: MSG_BUF_LEN(msg) is a constant.

Reply:
and lets flip orcer
__ubsan_report_error(caller, msg); -> __ubsan_report_error(kind, caller);